lib/alice/console.ex

```diff
@@ -2,25 +2,43 @@ defmodule Alice.Console do
   @moduledoc """
   Starts up the Console to allow testing of Alice's responses
   """
+
+  alias Alice.{Conn, Earmuffs, Router}
+
   @doc "Boots up the console and all associated constructs"
   def start() do
     setup_alice()
     IO.puts("Starting Alice Console")
-    repl()
+    repl(%{halt: false})
   end
 
-  defp repl(last_message \\ "")
-  defp repl("exit"), do: IO.puts("Goodbye!")
+  defp repl(%{halt: true}), do: IO.puts("Goodbye!")
 
-  defp repl(_last_message) do
+  defp repl(state) do
     message =
       IO.gets("alice> ")
       |> String.trim("\n")
       |> String.replace("@alice", "<@alice>")
 
-    Alice.HandlerCase.send_message(message)
+    conn = make_conn(message, state)
+    {:ok, state} = handle_message(conn)
     :timer.sleep(1)
-    repl(message)
+    repl(state)
+  end
+
+  defp handle_message(conn = %Conn{message: %{text: "exit"}}) do
+    {:ok, %{conn.state | halt: true}}
+  end
+
+  defp handle_message(conn = %Conn{}) do
+    conn =
+      cond do
+        Earmuffs.blocked?(conn) -> Earmuffs.unblock(conn)
+        Conn.command?(conn) -> Router.match_commands(conn)
+        true -> Router.match_routes(conn)
+      end
+
+    {:ok, conn.state}
```

Comment:
I love the changes to the console. As I'm looking at these methods though, they really look like a ChatBackend (which is a good thing). Perhaps the handle message and all the conn generation should be moved to a `Alice.ChatBackends.Console` module? Leaving this file as starting up the console and running the REPL? That new module could also be the basis for a new "ChatBackend" behaviour to marry it and Slack as well.

Reply:
Yeah probably but I had to draw the line somewhere with the refactoring so I just did the minimum to get state working between console commands. It'll probably move/all change in the genserver refactor I'm working on